src/main/scala/com/campudus/tableaux/database/model/FolderModel.scala

```diff
@@ -0,0 +1,178 @@
+      arr.get[String](4),   //created_at
+      arr.get[String](5)    //updated_at
+    )
+  }
+
+  implicit def convertLongToFolderId(id: Long): Option[FolderId] = {
+    //TODO not cool!
+    if (id == 0) {
+      None
+    } else {
+      Some(id.toLong)
+    }
+  }
+
+  implicit def convertStringToDateTime(str: String): Option[DateTime] = {
+    if (str == null) {
+      None
+    } else {
+      Option(DateTime.parse(str))
+    }
+  }
+
+  override def update(o: Folder): Future[Folder] = {
+    val update =
+      s"""UPDATE $table SET
+                         |name = ?,
+                         |description = ?,
+                         |idparent = ?,
+                         |updated_at = CURRENT_TIMESTAMP WHERE id = ? RETURNING created_at, updated_at""".stripMargin
+
+    for {
+      result <- connection.singleQuery(update, Json.arr(o.name, o.description, o.parent.orNull, o.id.get.toString))
+      resultArr <- Future.apply(updateNotNull(result))
+    } yield {
+      Folder(
+        o.id,
+        o.name,
+        o.description,
+        o.parent,
+        resultArr.head.get[String](0), //created_at
+        resultArr.head.get[String](1)  //updated_at
+      )
+    }
+  }
+
+  override def size(): Future[Long] = {
+    val select = s"SELECT COUNT(*) FROM $table"
+
+    for {
+      result <- connection.singleQuery(select, Json.emptyArr())
+      resultArr <- Future.apply(selectNotNull(result))
+    } yield {
+      resultArr.head.get[Long](0)
+    }
+  }
+
+  override def retrieveAll(): Future[Seq[Folder]] = {
+    val select =
+      s"""SELECT
+         |id,
+         |name,
+         |description,
+         |idparent,
+         |created_at,
+         |updated_at FROM $table""".stripMargin
+
+    for {
+      result <- connection.singleQuery(select, Json.emptyArr())
+      resultArr <- Future(getSeqOfJsonArray(result))
+    } yield {
+      resultArr.map { row =>
+        convertJsonArrayToFolder(row)
+      }
+    }
+  }
+
+  def retrieveSubfolders(id: FolderId): Future[Seq[Folder]] = {
+    val select =
+      s"""SELECT
+         |id,
+         |name,
+         |description,
+         |idparent,
+         |created_at,
+         |updated_at FROM $table WHERE idparent = ?""".stripMargin
+
+    for {
+      result <- connection.singleQuery(select, Json.arr(id))
+      resultArr <- Future(getSeqOfJsonArray(result))
+    } yield {
+      resultArr.map { row =>
+        convertJsonArrayToFolder(row)
+      }
+    }
+  }
+
+  override def delete(o: Folder): Future[Unit] = {
+    deleteById(o.id.get)
+  }
+
+  override def retrieve(id: FolderId): Future[Folder] = {
+    val select =
+      s"""SELECT
+         |id,
+         |name,
+         |description,
+         |idparent,
+         |created_at,
+         |updated_at FROM $table WHERE id = ?""".stripMargin
+
+    for {
+      result <- connection.singleQuery(select, Json.arr(id.toString))
+      resultArr <- Future.apply(selectNotNull(result))
+    } yield {
+      resultArr.head
```

Comment:
This looks magical to me. Would be better to not make `convertJsonArrayToFolder` implicit and use it here explicitly (like you did in `File`)